sefaria/helper/topic.py

```diff
@@ -1051,11 +1086,9 @@ def update_topic(topic_obj, **kwargs):
     old_category = ""
     orig_slug = topic_obj.slug
 
-    if 'title' in kwargs and kwargs['title'] != topic_obj.get_primary_title('en'):
-        topic_obj.add_title(kwargs['title'], 'en', True, True)
-
-    if 'heTitle' in kwargs and kwargs['heTitle'] != topic_obj.get_primary_title('he'):
-        topic_obj.add_title(kwargs['heTitle'], 'he', True, True)
+    update_topic_titles(topic_obj, kwargs)
+    if kwargs.get('catSlug') == 'authors':
+        topic_obj = update_authors_data(topic_obj, kwargs)
```

Comment:
if update_authors_data failed and update_topic_titles passed, would it create a mess in our data? It is difficult to account for every failure, but one thing we can account for is serializing and validating data before it gets to the functions. All the data that we expect to go to internal functions should be validated at the API level, and once it is clean and clear to go we can be much more assured that it won't fail